### DISCUSSION TITLE
Add support for CTranslate2 backend

### DISCUSSION THREAD

**Discussion by lewoudar** [outlines-dev/outlines#975] [Feature requests]
Hi everyone,
first of all, I want to thank the author for this nice library.
It would be interesting if the project supports [Ctranslate2](https://github.com/OpenNMT/CTranslate2/). It promises some huge performance improvements.

**Comment by rlouf** [comment]
We could, it looks like the model forward pass is exposed with the Python bindings: https://opennmt.net/CTranslate2/python/ctranslate2.Generator.html#ctranslate2.Generator.generate_tokens. Although that would represent more work than with e.g. the `llama.cpp` integration.
